### PR TITLE
Feature/mapcore widget

### PR DESCRIPTION
**Pull request by odeimaiz**
## What do these changes do?
This PR creates a new dynamic service that renders mapcore's flatmap widget.

For building the backend we:
- clone [ABI-Software/flatmapvuer](https://github.com/ABI-Software/flatmapvuer) repo
- fix it to a specific commit
- build/compile
- serve

## Related issue number
related to ITISFoundation/osparc-issues#24

https://github.com/ABI-Software/scaffoldvuer

image

## Checklist

- [ ] I think the code is well written
- [ ] I updated the table of contents (make toc)
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] **Runs in the swarm**
- [ ] If you design a new module, add your user to .github/CODEOWNERS

### CODE REVIEW

File: services/dy-mapcore-widget/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+1. TODO
+
+### dependencies
+
+TODO
+
+## production
+
+1. TODO
```
Comment (GitHK) [inline services/dy-mapcore-widget/README.md:15]: extra instructions would be helpful here to allow others to make changes

Comment (odeimaiz) [reply services/dy-mapcore-widget/README.md]: Done

**Comment by GitHK** [comment]
When trying to build the image locally, I get the following error:

```
Step 15/39 : FROM base AS production
 ---> f2659dfc9af3
Step 16/39 : COPY --from=base --chown=scu:scu $HOME/flatmapvuer $HOME/flatmapvuer
 ---> 69e432fc19ef
Step 17/39 : COPY --chown=scu:scu services/dy-mapcore-widget/inputs $HOME/inputs

ERROR: for mapcore-widget  (<Service: mapcore-widget>, 'COPY failed: stat /var/lib/docker/tmp/docker-builder235602545/services/dy-mapcore-widget/inputs: no such file or directory')
ERROR: Service 'mapcore-widget' failed to build: COPY failed: stat /var/lib/docker/tmp/docker-builder235602545/services/dy-mapcore-widget/inputs: no such file or directory
make: *** [Makefile:55: build] Error 1
```

**Comment by odeimaiz** [comment]
> When trying to build the image locally, I get the following error:
> 
> ```
> Step 15/39 : FROM base AS production
>  ---> f2659dfc9af3
> Step 16/39 : COPY --from=base --chown=scu:scu $HOME/flatmapvuer $HOME/flatmapvuer
>  ---> 69e432fc19ef
> Step 17/39 : COPY --chown=scu:scu services/dy-mapcore-widget/inputs $HOME/inputs
> 
> ERROR: for mapcore-widget  (<Service: mapcore-widget>, 'COPY failed: stat /var/lib/docker/tmp/docker-builder235602545/services/dy-mapcore-widget/inputs: no such file or directory')
> ERROR: Service 'mapcore-widget' failed to build: COPY failed: stat /var/lib/docker/tmp/docker-builder235602545/services/dy-mapcore-widget/inputs: no such file or directory
> make: *** [Makefile:55: build] Error 1
> ```
> 
> Looks like the same error inside the CI.

gitkeeps added

File: services/dy-mapcore-widget/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+RUN pip3 install --upgrade pip wheel setuptools
+
+# ----------------------------------------------------------------
+# install simcore-sdk
+RUN apk add --no-cache --virtual .build-deps \
+    g++ \
+    git \
+    python3-dev \
+    postgresql-dev && \
+    pip install --no-cache-dir git+https://github.com/ITISFoundation/osparc-simcore.git@master#subdirectory=packages/service-library &&\
```
Comment (pcrespov) [inline services/dy-mapcore-widget/Dockerfile:32]: please freeze these as well as the dependencies . Actually,  Do we really need them?

**Comment by GitHK** [comment]
👍

To be sure this is working, would it be possible to add a test which starts the entire thing and requests a feature on the ui to check it is working? (like puppeteer does for the main portal)
The current tests make no sense to me.

File: services/dy-mapcore-widget/Dockerfile (hunk 2 above)
Comment (odeimaiz) [reply services/dy-mapcore-widget/Dockerfile:32]: This is the result of a copy-pasting, I don't even know if we need them.

It works for me, if you want to clean it up, feel free to take it over.

Comment (pcrespov) [reply services/dy-mapcore-widget/Dockerfile:32]: :-(

Comment (odeimaiz) [reply services/dy-mapcore-widget/Dockerfile:32]: https://github.com/ITISFoundation/osparc-issues/issues/256

**Pull request merged**